### PR TITLE
Test whether http client span ends after headers or body is received

### PR DESCRIPTION
**Pull request by laurit**
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11726 added a test that verifies that http client span is ended after body is received. This PR moves that test to the common http client test. Also a test that verifies that http client span is ended after headers are received is added. Only one of these tests should pass on given http client.

### CODE REVIEW

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java [hunk 1]
```diff
@@ -955,6 +956,69 @@ void highConcurrencyOnSingleConnection() {
...
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> assertClientSpan(span, uri, method, 200, null).hasNoParent(),
+              span -> assertServerSpan(span).hasParent(trace.getSpan(0)));
+          SpanData span = trace.getSpan(0);
+          // make sure the span is at least as long as the delay we set when sending the request
+          assertThat(
+                  span.getEndEpochNanos() - span.getStartEpochNanos()
+                      >= TimeUnit.SECONDS.toNanos(1))
```
Comment (jaydeluca) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java:984]: It looks like we are setting a delay of 1 millisecond, but asserting that the span is longer than 1 nanosecond. Should those match?

Comment (laurit) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java:984]: Perhaps you misread this `TimeUnit.SECONDS.toNanos(1)` returns 1 second in nanos = 1000000000

File: instrumentation/google-http-client-1.19/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/AbstractGoogleHttpClientTest.java [hunk 2]
```diff
@@ -83,7 +83,9 @@ public int sendRequest(HttpRequest request, String method, URI uri, Map<String,
       throws Exception {
     HttpResponse response = sendRequest(request);
     // read request body to avoid broken pipe errors on the server side
-    response.parseAsString();
+    if (!uri.toString().contains("/long-request")) {
+      response.parseAsString();
```
Comment (trask) [inline instrumentation/google-http-client-1.19/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/AbstractGoogleHttpClientTest.java:87]: can you add a comment why skipping this check for `/long-request`?

File: instrumentation/http-url-connection/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionUseCachesFalseTest.java [hunk 3]
```diff
@@ -45,7 +45,9 @@ public int sendRequest(
       Span parentSpan = Span.current();
       InputStream stream = connection.getInputStream();
       assertThat(Span.current()).isEqualTo(parentSpan);
-      readLines(stream);
+      if (!uri.toString().contains("/long-request")) {
+        readLines(stream);
```
Comment (trask) [inline instrumentation/http-url-connection/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionUseCachesFalseTest.java:49]: and here

**Review by trask — APPROVED** [review]

File: instrumentation/google-http-client-1.19/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/AbstractGoogleHttpClientTest.java (hunk 2 above)
Comment (laurit) [reply instrumentation/google-http-client-1.19/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/AbstractGoogleHttpClientTest.java:88]: Initially I used a longer delay that caused reading response to fail. Now this just makes the test a bit faster. I added a comment, but removing this `if` would also be fine.

**Review by trask — APPROVED** [review]

**Pull request merged**